src/ape/managers/config.py

```diff
@@ -69,7 +82,27 @@ def _plugin_configs(self) -> Dict[str, ConfigItem]:
         # Top level config items
         self.name = user_config.pop("name", "")
         self.version = user_config.pop("version", "")
-        self.dependencies = user_config.pop("dependencies", {})
+
+        dependencies = user_config.pop("dependencies", [])
+        if not isinstance(dependencies, list):
+            raise ConfigError("'dependencies' config item must be a list of dicts.")
+
+        decoded_deps = []
+        dep_config_cls: Union[Type[IPFSDependency], Type[GithubDependency]]
+        for dep in dependencies:
+            if "ipfs" in dep:
+                dep_config_cls = IPFSDependency
+            elif "github" in dep:
+                dep_config_cls = GithubDependency
```

Comment:
check out this: https://pydantic-docs.helpmanual.io/usage/model_config/#smart-union

Reply:
Looks like smart union does not support compound types yet, and since this is a `List[DependencyType]` it might not work yet. Looks Pydantic plans to support that eventually though.